controllers/storagecluster/storageclient.go

```diff
@@ -67,6 +75,14 @@ func (s *storageClient) ensureCreated(r *StorageClusterReconciler, storagecluste
 
 		controllerutil.AddFinalizer(storageClient, internalComponentFinalizer)
 
+		if storagecluster.Spec.Network != nil && storagecluster.Spec.Network.IsMultus() {
+			cephNWAnnotationValue, err := getCephNetworkAnnotationValue(storagecluster.Spec.Network, storagecluster.Namespace)
+			if err != nil {
+				return fmt.Errorf("failed to get Ceph network annotation value: %v", err)
+			}
+			util.AddAnnotation(storageClient, cniNetworksAnnotationKey, cephNWAnnotationValue)
+		}
+
```

Comment:
Can this please be moved above the AddFinalizer so we keep the add finalizer as the last change we do within the create or update?

Reply:
moved above AddFinalizer